### PR TITLE
Fix group creation issue

### LINKED ISSUE #65: Group creation requires ses_user_enabled to be true
## Describe the Bug

We are trying to create a new ses group by setting "ses_user_enabled" to false and "ses_group_enabled" to true in our "ses" module. However, we get this error when applying the changes.

```
│ Error: "policy" contains an invalid JSON policy
│ 
│   with module.ses.aws_iam_group_policy.ses_group_policy[0],
│   on .terraform/modules/ses/main.tf line 70, in resource "aws_iam_group_policy" "ses_group_policy":
│   70:   policy = join("", data.aws_iam_policy_document.ses_policy.*.json)
```

It refers to data.aws_iam_policy_document:

```
data "aws_iam_policy_document" "ses_policy" {
  count = (module.this.enabled && var.ses_user_enabled) ? 1 : 0
                                                                                                          
  statement {                                                                                             
    actions   = var.iam_permissions                                                                       
    resources = concat(aws_ses_domain_identity.ses_domain.*.arn, var.iam_allowed_resources)               
  }                                                                                                       
} 
```

The count is there to enable or disable this policy document but it only enables it if "var.ses_user_enabled" is true (we want it "false" to create only the ses group). 

## Expected Behavior

A new group can be created without the need to set "ses_user_enabled" to true. As a workaround, we replaced the "count" line with this:

`count = (module.this.enabled && (var.ses_user_enabled || local.create_group_enabled)) ? 1 : 0`

## Steps to Reproduce

Steps to reproduce the behavior: Set "ses_user_enabled" to false and "ses_group_enabled" to true in the "ses" module then run "terraform apply".

https://github.com/cloudposse/terraform-aws-ses/pull/66

### PR DESCRIPTION
**Pull request by ncjmlnchth**
## what
* We are trying to create a new ses group by setting "ses_user_enabled" to false and "ses_group_enabled" to true in our "ses" module. However, we get this error when applying the changes: 

```
│ Error: "policy" contains an invalid JSON policy
│ 
│   with module.ses.aws_iam_group_policy.ses_group_policy[0],
│   on .terraform/modules/ses/main.tf line 70, in resource "aws_iam_group_policy" "ses_group_policy":
│   70:   policy = join("", data.aws_iam_policy_document.ses_policy.*.json)
```

* It refers to data.aws_iam_policy_document:

```
data "aws_iam_policy_document" "ses_policy" {
  count = (module.this.enabled && var.ses_user_enabled) ? 1 : 0
                                                                                                          
  statement {                                                                                             
    actions   = var.iam_permissions                                                                       
    resources = concat(aws_ses_domain_identity.ses_domain.*.arn, var.iam_allowed_resources)               
  }                                                                                                       
} 
```

* The count is there to enable or disable this policy document but it only enables it if "var.ses_user_enabled" is true (we want it "false" to create only the ses group). As a workaround, we replaced the "count" line with this:

`count = (module.this.enabled && (var.ses_user_enabled || local.create_group_enabled)) ? 1 : 0`

## why
* Enable group creation without the requirement of setting "ses_user_enabled" to true.

## references
* Closes #65

### CODE REVIEW

**Comment by stephan242** [comment]
@max-lobur sorry for nagging, but any chance you could have a quick look? Just a one line change.

File: main.tf [hunk 1]
```diff
@@ -46,7 +46,7 @@ locals {
 }
 
 data "aws_iam_policy_document" "ses_policy" {
-  count = (module.this.enabled && var.ses_user_enabled) ? 1 : 0
+  count = (module.this.enabled && (var.ses_user_enabled || local.create_group_enabled)) ? 1 : 0
```
Comment (goruha) [inline main.tf:49]:
```suggestion
  count = local.create_user_enabled || local.create_group_enabled ? 1 : 0
```

**Review by goruha — CHANGES_REQUESTED** [review]

File: main.tf (hunk 1 above)
Comment (goruha) [reply main.tf:49]: @stephan242 can we improve the condition this way?

**Comment by ncjmlnchth** [comment]
@goruha PR's been update with your suggestion.

**Review by goruha — APPROVED** [review]

**Comment by goruha** [comment]
@ncjmlnchth LGTM
Thanks for your contribution.

**Pull request merged**